### PR TITLE
[[ Bug 18175 ]] Warn if saving in a legacy format causes data loss

### PR DESCRIPTION
**Pull request by montegoulding**
Requires https://github.com/livecode/livecode/pull/4365

### CODE REVIEW

File: Toolset/libraries/revidelibrary.8.livecodescript [hunk 1]
```diff
@@ -7294,16 +7301,29 @@ on revIDESaveStack pStackID
...
+    global gRevStackStatus
+  
+   if revIDEGetPreference("cPreserveStackVersion") is true then
       put revIDEStackFileVersion(tStackFilename) into tStackFileVersion
+      
+      local tFileVersions
+      put tStackFileVersion into tFileVersions[1]
+      put the minStackFileVersion of stack tStackName into tFileVersions[2]
+      if highestVersionNumber(tStackFileVersion, tFileVersions[2]) is 2 then
+         answer revIDELocalise("The preference to preserve stack file formats is is set, however, saving this stack file as version %1 will result in data loss. The minimum stack file version for this stack is %2.", tFileVersions) with revIDELocalise("Save as %1", tFileVersions) and  revIDELocalise("Save as %2", tFileVersions) and revIDELocalise("Cancel")
```
Comment (livecodeali) [inline Toolset/libraries/revidelibrary.8.livecodescript]: Could you split this line up a bit please?

File: Toolset/libraries/revidelibrary.8.livecodescript [hunk 2]
```diff
@@ -7576,13 +7593,17 @@ on revIDEActionSaveStackAs pStackID, pSubstackOnly
...
-   # Display warning if user has not selected the latest file format
-   if tLegacy then
-      answer revIDELocalise("Saving in legacy format may result in loss of data if you are using recent LiveCode features. Are you sure you wish to continue?") with revIDELocalise("No") and revIDELocalise("Yes")
-      if it is not revIDELocalise("Yes") then
+   
+   local tFileVersions
+   put tVersion into tFileVersions[1]
+   put the minStackFileVersion of stack tShortName into tFileVersions[2]
+   if highestVersionNumber(tVersion, tFileVersions[2]) is 2 then
+      answer revIDELocalise("Saving this stack file as version %1 will result in data loss. The minimum stack file version for this stack is %2.", tFileVersions) with revIDELocalise("Save as %1", tFileVersions) and  revIDELocalise("Save as %2", tFileVersions) and revIDELocalise("Cancel")
```
Comment (livecodeali) [inline Toolset/libraries/revidelibrary.8.livecodescript]: And this one!

File: Toolset/libraries/revidelibrary.8.livecodescript (hunk 1 above)
Comment (montegoulding) [inline Toolset/libraries/revidelibrary.8.livecodescript]: will do!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 3edadd0
@montegoulding Thanks!

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 3edadd041efd3e07c74329e04cf9934c9d755378

**Pull request merged**